docs/content/latest/explore/ysql-language-features/partitions.md

```diff
@@ -17,29 +17,28 @@ This section describes how to partition tables in YugabyteDB using YSQL.
+YSQL allows you to specify how exactly to divide a table. You provide a partitioning method and partition key consisting of a list of columns or expressions. The divided table is called a partitioned table, and the resulting tables are called partitions. When you insert rows into a partitioned table, they are redirected to a partition depending on the value of the partition key. You can also directly insert rows into the partition table itself, and those rows can be fetched by querying the parent table.
 
-You can partition partitions, in which case they woud have their own distinct indexes, constraints, and default values.
+You can nest partitions, in which case they woud have their own distinct indexes, constraints, and default values.
 
-A regular table cannot become a partitioned table, just as a partitioned table cannot become a regular table. That said, you can add a regular or partitioned table that has data as a partition of a partitioned table, and you can remove a partition from a partitioned table making it a standalone table.
+A regular table cannot become a partitioned table, just as a partitioned table cannot become a regular table. That said, YSQL allows attaching a regular table (provided it has the same schema as that of the partitioned table) as a partition to the partitioned  table. Conversely, a partition can be detached from the partitioned table, in which case it can behave as a regular table that is not part of the partitioning hierarchy.
 
 A partitioned table and its partitions have hierarchical structure and are subject to its rules, except the following:
 
-- If there are no partitions, you can use the `ONLY` clause to add or drop a constraint on the partitioned table. If there are partitions, you cannot use `ONLY` because adding or deleting constraints on the  partitioned tableis not supported in such cases.
-- Partitions cannot have columns which do not exist in their partitioned table. You can add a partition using `ALTER TABLE ... ATTACH PARTITION` if their columns match the partitioned table, including a `oid` column.
+- If there are no partitions, you can use the `ONLY` clause to add or drop a constraint on the partitioned table. If there are partitions, you cannot use `ONLY` because adding or deleting constraints on the  partitioned table is not supported in such cases.
+- Partitions cannot have columns which do not exist in the partitioned table. You can add a partition using `ALTER TABLE ... ATTACH PARTITION` if their columns match the partitioned table, including a `oid` column.
 - Partitions inherit `CHECK` and `NOT NULL` constraints of their partitioned table. This does not include `CHECK` constraints marked `NO INHERIT`.
```

Comment:
You cannot create constraints marked NO INHERIT at all. Please check the documentation of Postgres 11.2 before adding statements here. It's also very important to add test cases (or log GH issues requesting for tests to be added) for all statements mentioned in the docs, otherwise it is very embarrassing for the company to document a behavior that is not supported. 

"yugabyte=# alter table plt1_e add constraint cons check (a > 0) no inherit;
ERROR:  cannot add NO INHERIT constraint to partitioned table "plt1_e"

Reply:
Correct, and that's what we have in the doc. Perhaps the sentence wasn't as clear as it could have been, so I modified it a little:

A partitioned table and its partitions have hierarchical structure and are subject to its rules, except the following:
- If there are no partitions, you can use the `ONLY` clause to add or drop a constraint on the partitioned table. If there are partitions, you cannot use `ONLY` because adding or deleting constraints on the  partitioned table is not supported in such cases.
- You can add a regular table to the partition hierarchy using `ALTER TABLE ... ATTACH PARTITION` if their columns match the partitioned table, including a `oid` column.
- Partitions inherit `CHECK` and `NOT NULL` constraints of the partitioned table, with the following exceptions: 
  - `CHECK` constraints marked `NO INHERIT`be cannot be created on partitioned tables.
  - The `NOT NULL` constraint cannot be added to a partition's column if the constraint is present in their partitioned table.